docs/contributing.rst

```diff
@@ -343,56 +352,20 @@ This should generate documentation in the ``docs/_build/html``
 directory.
 
 
-Other methods for running the client
-====================================
-
-Vagrant
--------
-
-If you are a Vagrant user, Certbot comes with a Vagrantfile that
-automates setting up a development environment in an Ubuntu 14.04
-LTS VM. To set it up, simply run ``vagrant up``. The repository is
-synced to ``/vagrant``, so you can get started with:
-
-.. code-block:: shell
-
-  vagrant ssh
-  cd /vagrant
-  sudo ./venv/bin/certbot
-
-Support for other Linux distributions coming soon.
-
-.. note::
-   Unfortunately, Python distutils and, by extension, setup.py and
-   tox, use hard linking quite extensively. Hard linking is not
-   supported by the default sync filesystem in Vagrant. As a result,
-   all actions with these commands are *significantly slower* in
-   Vagrant. One potential fix is to `use NFS`_ (`related issue`_).
-
-.. _use NFS: http://docs.vagrantup.com/v2/synced-folders/nfs.html
-.. _related issue: https://github.com/ClusterHQ/flocker/issues/516
-
-
-Docker
-------
-
-OSX users will probably find it easiest to set up a Docker container for
-development. Certbot comes with a Dockerfile (``Dockerfile-dev``)
-for doing so. To use Docker on OSX, install and setup docker-machine using the
-instructions at https://docs.docker.com/installation/mac/.
-
-To build the development Docker image::
+.. _docker:
 
-  docker build -t certbot -f Dockerfile-dev .
+Running the client with Docker
+==============================
 
-Now run tests inside the Docker image:
-
-.. code-block:: shell
+macOS users will probably find it easiest to use Docker Compose for
+development. To install Docker Compose on macOS, follow the instructions at
+https://docs.docker.com/docker-for-mac/install/.
 
-  docker run -it certbot bash
-  cd src
-  tox -e py27
+Now you can develop locally and run Certbot and test your changes in Docker.
+For example, from anywhere inside your clone of the Certbot repository you can
+run the following to check for linting errors::
 
+  docker-compose run --rm --service-ports development bash -c 'cd src;tox -e lint'
```

Comment:
> docker-compose creates a docker image that spits you out into a top-level where there's a subdirectory src? maybe we want to note this?

I'd ideally just change it so you're automatically in `src` which Jacob and I discussed earlier in IRC. I'll make that one line change.

> does docker-compose have a reference to the repo on the host, or is copied somewhere when you create an image? so, could you run with run and exec to do things interactively?

The repo on the host is mounted inside of the Docker container so you can do things interactively. What would you like to see added here?